x-pack/plugins/observability/public/pages/slos_page/hooks/use_fetch_slo_list.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useMemo } from 'react';
+import { HttpSetup } from '@kbn/core/public';
+
+import type { SLO, SLOList } from '../../../typings/slo';
+import { useDataFetcher } from '../../../hooks/use_data_fetcher';
+
+const EMPTY_LIST = {
+  results: [],
+  total: 0,
+  page: 0,
+  perPage: 0,
+};
+
+export const useFetchSloList = (): [boolean, SLOList] => {
+  const params = useMemo(() => ({}), []);
+  const shouldExecuteApiCall = useCallback((apiCallParams: {}) => true, []);
+
+  const { loading, data: sloList } = useDataFetcher<{}, SLOList>({
+    paramsForApiCall: params,
+    initialDataState: EMPTY_LIST,
+    executeApiCall: fetchSloList,
+    shouldExecuteApiCall,
+  });
+
+  return [loading, sloList];
+};
+
+const fetchSloList = async (
+  params: {},
+  abortController: AbortController,
+  http: HttpSetup
+): Promise<SLOList> => {
+  try {
+    const response = await http.get<Record<string, unknown>>(`/api/observability/slos`, {
+      query: {},
+      signal: abortController.signal,
+    });
+    if (response !== undefined) {
+      return toSLOList(response);
+    }
+  } catch (error) {
+    // ignore error for retrieving slos
+  }
+
+  return EMPTY_LIST;
+};
+
+function toSLOList(response: Record<string, unknown>): SLOList {
+  if (!Array.isArray(response.results)) {
+    throw new Error('Invalid response');
+  }
+
+  return {
+    results: response.results.map((result) => toSLO(result)),
+    page: Number(response.page),
+    perPage: Number(response.per_page),
+    total: Number(response.total),
+  };
+}
+
+function toSLO(result: any): SLO {
```

Comment:
Why do we get result as `any`? Since we implemented the API, shouldn't we have some assumptions about the response?
Another question: Why didn't you use `io-ts` to validate the API response?

Reply:
At first It was typed as unknown, but this is very annoying to work with as you have to safeguard against every.nested.object.property. Thus, I changed for any 🙈 

Ideally we should use `io-ts` to validate the response at runtime I agree. 
Do you think we can import things from the `server` into the `public` folder? 

Same for the type, I'd like to export the rest_specs/schema from the server so we can build the frontend version of the SLO model. I don't want to leak the server's domain models though as this would couple backend and frontend in a strict way.

Reply:
> Do you think we can import things from the `server` into the `public` folder?

I don't know, my guess is we cannot.
@weltenwort How do we usually check the types between client and server in Kibana?

Anyway, I think we can improve it later :)

Reply:
@maryam-saeidi, you're guessing correctly. Imports between client and server code are not allowed. Shared code can be either put into a package or into a plugin's `common` folder, which can be import from both client and server.